src/huntsman/pocs/camera/zwo.py

```diff
@@ -177,6 +183,27 @@ def reconnect(self):
         self._reset_usb()
         return self.connect()
 
+    def take_exposure(self, defocused=False, *args, **kwargs):
+        """ Overrride class method to add defocusing offset. """
+        if defocused:
+            required_focus_offset = self._defocus_offset - self._current_defocus_offset
+            self.logger.debug(f"Applying focus offset of {self._defocus_offset} for defocused "
+                              "exposure.")
+        else:
+            self.logger.debug("Nullifying defocus offset.")
+            required_focus_offset = -self._current_defocus_offset
```

Comment:
as `defocused=False` won't this be run with every exposure? Do you wanna detect if `self._current_defocus_offset` has been set here before subtracting off a value?

Reply:
`self._current_defocus_offset` has a default value of 0. This line will always work regardless of how many times it is called.